File: app/src/main/java/starbright/com/projectegg/util/TextViewRecyclerAdapter.java
package starbright.com.projectegg.util;

import android.content.Context;
import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import java.util.List;

import butterknife.BindView;
import butterknife.ButterKnife;
import starbright.com.projectegg.R;

public class TextViewRecyclerAdapter extends RecyclerView.Adapter<TextViewRecyclerAdapter.ViewHolder> {

    private Context mContext;
    private List<String> mDataSource;

    public TextViewRecyclerAdapter(Context context, List<String> dataSource) {
        mContext = context;
        mDataSource = dataSource;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(mContext)
                .inflate(R.layout.item_text_only, parent, false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        holder.tvText.setText(mDataSource.get(position));
    }

    @Override
    public int getItemCount() {
        return mDataSource.size();
    }

    class ViewHolder extends RecyclerView.ViewHolder {

        @BindView(R.id.tv_text)
        TextView tvText;

        ViewHolder(View itemView) {
            super(itemView);
            ButterKnife.bind(this, itemView);
        }
    }
}

File: app/src/main/java/starbright/com/projectegg/util/TextViewRecyclerAdapter.kt
/*
 * Copyright (c) by Andreas (oentoro.andreas@gmail.com)
 * created at 25 - 7 - 2020.
 */

/**
 * Created by Andreas on 19/8/2018.
 */

/**
 * Created by Andreas on 19/8/2018.
 */

/**
 * Created by Andreas on 19/8/2018.
 */

package starbright.com.projectegg.util

import starbright.com.projectegg.databinding.ItemTextOnlyBinding
import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import starbright.com.projectegg.R

class TextViewRecyclerAdapter(private val mContext: Context, private val mDataSource: List<String>) : RecyclerView.Adapter<TextViewRecyclerAdapter.ViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val binding = ItemTextOnlyBinding.inflate(
            LayoutInflater.from(mContext), parent, false
        )
        return ViewHolder(binding)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        holder.binding.tvText.text = mDataSource[position]
    }

    override fun getItemCount(): Int {
        return mDataSource.size
    }

    inner class ViewHolder(val binding: ItemTextOnlyBinding) : RecyclerView.ViewHolder(binding.root) {

    }
}
